epochStart/metachain/systemSCs.go

```diff
@@ -284,15 +254,15 @@ func (s *systemSCProcessor) ProcessSystemSmartContract(
 		}
 	}
 
-	if s.flagESDTEnabled.IsSet() {
+	if s.enableEpochsHandler.IsESDTFlagEnabledForCurrentEpoch() {
```

Comment:
remove the `CurrentEpoch` from the naming? applicable for all changes, so I won't leave too many comments